docs/parameters.md

```diff
@@ -0,0 +1,322 @@
+Instead of specifying a reference to a single `Secret` (or `ConfigMap`) key, you 
+just set the reference to the `Secret` (or `ConfigMap`), and all keys will be 
+automatically processed as separate parameters.
+
+In other words, instead of 
+```yaml
+  ...
+  parameters:
+    - name: key1
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: key1
+    - name: key2
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: key2
+     ...
+```
+you can just set it once in a `parametersFrom` field:
+```yaml
+  ...
+  parametersFrom:
+    - secretRef:
+        name: mysecret
+```
+
+You can also pass several `Secret`s and/or `ConfigMap`s at once:
+```yaml
+  ...
+  parametersFrom:
+    - secretRef:
+        name: secret1
+    - secretRef:
+        name: secret2
+    - configMapRef:
+        name: configmap1
+```
+
+Parameter values could be passed as JSON objects, the same way as for [individual parameters](#json-object):
+```yaml
+  ...
+  parametersFrom:
+    - type: json
+      secretRef:
+        name: mysecret
+```
+
+### Conflict resolution
+
+When a parameter name exists in multiple sources defined in `parametersFrom` or `parameters` 
+field, the value associated with the last source will take precedence.
+Values defined in `parameters` with a duplicate name will take precedence over `parametersFrom`.
+This conflict resolution strategy is consistent with the one supported by Kubernetes
+for environment variables (`env` and `envFrom` fields) in `Pod` resource.
+
+For example, for the structure like this
+```yaml
+  ...
+  parameters:
+    - name: param
+      value: 1
+    - name: param
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: key
+    - name: param
+      value: 2
+  parametersFrom:
+    - name: param
+      secretRef:
+        name: mysecret
+```
+the parameters JSON structure to be sent to the broker is going to be
+```json
+{
+  "param": "2"
+}
+```
+
+To avoid conflicts for `parametersFrom` (for example, in the case when multiple 
+secrets of the same structure need to be passed to the broker), the user can specify 
+a `name` field which will produce an extra nesting level.
+
+For example, let's say we have the following structure in a `Secret`:
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mysecret
+type: Opaque
+data:
+  username: YWRtaW4=
+  password: MWYyZDFlMmU2N2Rm
+```
+
+To inject credentials from two instances of such a `Secret`, we can avoid the 
+naming conflict by providing unique names:
+```yaml
+  ...
+  parametersFrom:
+    - name: master
+      secretRef:
+        name: secret1
+    - name: slave
+      secretRef:
+        name: secret2
+```
+which will result in the following JSON parameters object:
+```json
+{
+  "master": {
+    "username": "root",
+    "password": "letmein"
+  },
+  "slave": {
+    "username": "foo",
+    "password": "bar"
+  }
+}
+
+```
```

Comment:
Can we get a full listing all of the options in one spot so we can see the complete picture?

Also, it might be good to show a "type:json" example where the json/value is passed as a string. 
E.g.
```
parameters:
  - name: creds
    type: json
    value: { "id": 5, "name": "john", "password": "letmein" }
```